File: sample-videochat-conference-java/src/main/java/com/quickblox/sample/videochat/conference/java/adapters/CheckboxUsersAdapter.java
package com.quickblox.sample.videochat.conference.java.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.TextView;

import com.quickblox.sample.videochat.conference.java.R;
import com.quickblox.sample.videochat.conference.java.utils.UiUtils;
import com.quickblox.users.model.QBUser;

import java.util.ArrayList;
import java.util.List;


public class CheckboxUsersAdapter extends BaseAdapter {

    private QBUser currentUser;
    private List<Integer> initiallySelectedUsers;
    private List<QBUser> selectedUsers;
    private LayoutInflater inflater;
    private Context context;
    private List<QBUser> userList;

    public CheckboxUsersAdapter(Context context, List<QBUser> users, QBUser currentUser) {
        this.context = context;
        userList = users;
        inflater = LayoutInflater.from(context);
        this.currentUser = currentUser;
        selectedUsers = new ArrayList<>();
        selectedUsers.add(currentUser);
        initiallySelectedUsers = new ArrayList<>();
    }

    @Override
    public int getCount() {
        return userList != null ? userList.size() : 0;
    }

    @Override
    public QBUser getItem(int position) {
        return userList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    public void updateList(List<QBUser> newData) {
        userList = newData;
        notifyDataSetChanged();
    }

    public void add(QBUser item) {
        userList.add(item);
        notifyDataSetChanged();
    }

    public List<QBUser> getList() {
        return userList;
    }

    public void addList(List<QBUser> items) {
        userList.addAll(0, items);
        notifyDataSetChanged();
    }

    public void remove(QBUser item) {
        userList.remove(item);
        notifyDataSetChanged();
    }

    public void addSelectedUsers(List<Integer> userIds) {
        for (QBUser user : userList) {
            for (Integer id : userIds) {
                if (user.getId().equals(id)) {
                    selectedUsers.add(user);
                    initiallySelectedUsers.add(user.getId());
                    break;
                }
            }
        }
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        final QBUser user = getItem(position);

        final ViewHolder holder;
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.list_item_user, parent, false);
            holder = new ViewHolder();
            holder.userImageView = convertView.findViewById(R.id.image_user);
            holder.loginTextView = convertView.findViewById(R.id.text_user_login);
            holder.userCheckBox = convertView.findViewById(R.id.checkbox_user);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        if (isUserMe(user)) {
            holder.loginTextView.setText(context.getString(R.string.placeholder_username_you, user.getFullName()));
        } else {
            holder.loginTextView.setText(user.getFullName());
        }

        if (isAvailableForSelection(user)) {
            holder.loginTextView.setTextColor(context.getResources().getColor(R.color.text_color_black));
        } else {
            holder.loginTextView.setTextColor(context.getResources().getColor(R.color.text_color_medium_grey));
        }

        holder.userImageView.setBackgroundDrawable(UiUtils.getColorCircleDrawable(position));
        holder.userCheckBox.setVisibility(View.GONE);

        convertView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (!isAvailableForSelection(user)) {
                    return;
                }

                holder.userCheckBox.setChecked(!holder.userCheckBox.isChecked());
                if (holder.userCheckBox.isChecked()) {
                    selectedUsers.add(user);
                } else {
                    selectedUsers.remove(user);
                }
            }
        });

        holder.userCheckBox.setVisibility(View.VISIBLE);
        holder.userCheckBox.setChecked(selectedUsers.contains(user));

        return convertView;
    }

    protected boolean isUserMe(QBUser user) {
        return currentUser != null && currentUser.getId().equals(user.getId());
    }

    public List<QBUser> getSelectedUsers() {
        return selectedUsers;
    }

    protected boolean isAvailableForSelection(QBUser user) {
        return currentUser == null || !currentUser.getId().equals(user.getId()) && !initiallySelectedUsers.contains(user.getId());
    }

    protected static class ViewHolder {
        ImageView userImageView;
        TextView loginTextView;
        CheckBox userCheckBox;
    }
}

File: sample-videochat-conference-kotlin/src/main/java/com/quickblox/sample/videochat/conference/kotlin/adapter/CheckboxUsersAdapter.kt
package com.quickblox.sample.videochat.conference.kotlin.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.quickblox.sample.videochat.conference.kotlin.R
import com.quickblox.sample.videochat.conference.kotlin.utils.getColor
import com.quickblox.sample.videochat.conference.kotlin.utils.getColorCircleDrawable
import com.quickblox.users.model.QBUser
import kotlinx.android.synthetic.main.list_item_user.view.*


class CheckboxUsersAdapter(val context: Context, private var usersList: List<QBUser>,
                           private val currentUser: QBUser?) : RecyclerView.Adapter<CheckboxUsersAdapter.ViewHolder>() {

    private val selectedUsers: MutableList<QBUser> = ArrayList()

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val user = usersList[position]

        if (isUserMe(user)) {
            holder.loginTextView.text = context.getString(R.string.placeholder_username_you, currentUser?.fullName)
            selectedUsers.add(currentUser as QBUser)
        } else {
            holder.loginTextView.text = user.fullName
        }

        if (isAvailableForSelection(user)) {
            holder.loginTextView.setTextColor(getColor(R.color.text_color_black))
        } else {
            holder.loginTextView.setTextColor(getColor(R.color.text_color_medium_grey))
        }

        holder.userImageView.setBackgroundDrawable(getColorCircleDrawable(position))
        holder.userCheckBox.isChecked = selectedUsers.contains(user)

        holder.rootLayout.setOnClickListener {
            if (isAvailableForSelection(user)) {
                holder.userCheckBox.isChecked = !holder.userCheckBox.isChecked
                if (holder.userCheckBox.isChecked) {
                    selectedUsers.add(user)
                } else {
                    selectedUsers.remove(user)
                }
            }
        }
    }

    override fun getItemCount(): Int {
        return usersList.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(context).inflate(R.layout.list_item_user, parent, false))
    }

    private fun isUserMe(user: QBUser): Boolean {
        return currentUser != null && currentUser.id == user.id
    }

    private fun isAvailableForSelection(user: QBUser): Boolean {
        return currentUser == null || currentUser.id != user.id
    }

    fun getSelectedUsers(): List<QBUser> {
        return selectedUsers
    }

    fun updateUsers(usersList: List<QBUser>) {
        this.usersList = usersList
        selectedUsers.clear()
        notifyDataSetChanged()
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val userImageView = view.image_user
        val loginTextView = view.text_user_login
        val userCheckBox = view.checkbox_user
        val rootLayout = view.linear_layout_root
    }
}